clang/test/SemaTemplate/deduction-guide.cpp

```diff
@@ -335,3 +335,73 @@ namespace TTP {
 // CHECK-NEXT:      `-TemplateArgument type 'T':'type-parameter-0-0'{{$}}
 // CHECK-NEXT:        `-TemplateTypeParmType {{.+}} 'T' dependent depth 0 index 0{{$}}
 // CHECK-NEXT:          `-TemplateTypeParm {{.+}} 'T'{{$}}
+
+namespace GH64625 {
+
+template <class T> struct X {
```

Comment:
Can you add a test with multiple array members? like

```
template <class T> struct X {
  T t1[2];
  T t2[3];
};

X x = {{1, 2}, {3, 4, 5}};
```

Reply:
Oops, `InitListChecker` seemingly doesn't handle such a case very well. It would produce only one element, namely `T (&&)[2]` atm. Will look into that, and thanks for spotting it!

Reply:
So, I realized the problem: we added the first parameter of the type `T (&&)[2]` but the following logic consumed all the remaining initializers by altering the `Index` - it's then hard and involved to retain the 'no-elision' version of parameters in one go of an `InitListChecker`. I decided to decouple it into two separate calls now.